client/remoterepo.c

```diff
@@ -368,12 +369,19 @@ TDNFCheckRepoMDFileHashFromMetalink(
         hashInfo = currHashInfo;
     }
 
-    dwError = TDNFChecksumFromHexDigest(hashInfo->value, digest);
-    BAIL_ON_TDNF_ERROR(dwError);
-
-    dwError = TDNFCheckHash(pszFile, digest, hash_Type);
-    BAIL_ON_TDNF_ERROR(dwError);
+    if (hashInfo != NULL)
+    {
+        dwError = TDNFChecksumFromHexDigest(hashInfo->value, digest);
+        BAIL_ON_TDNF_ERROR(dwError);
 
+        dwError = TDNFCheckHash(pszFile, digest, hash_Type);
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+    else
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
```

Comment:
should we put more accurate error here?

Reply:
Changing it to `ERROR_TDNF_INVALID_REPO_FILE` - I think that's better, from looking at the code preceding this.